src/modules/stream/internal.js

```diff
@@ -73,6 +75,36 @@ async function handleYoutubeStream(streamInfo, res) {
     }
 }
 
+function transformHLSMediaPlaylist(streamInfo, hlsPlaylist) {
+    function generateInternalStreamsOfSegments(segment) {
+        const fullUri = new URL(segment.uri, streamInfo.url).toString();
+        segment.uri = createInternalStream(fullUri, streamInfo);
+
+        return segment;
+    }
+
+    hlsPlaylist.segments =
+        hlsPlaylist.segments.map(generateInternalStreamsOfSegments);
+    hlsPlaylist.prefetchSegments =
+        hlsPlaylist.prefetchSegments.map(generateInternalStreamsOfSegments);
+
+    return hlsPlaylist;
+}
+
+async function handleHLSPlaylist(streamInfo, req, res) {
```

Comment:
please follow camelcase for abbreviations, as it's more readable this way: `handleHlsPlaylist`